Source/EventFlow/Subscribers/DispatchToEventSubscribers.cs

```diff
@@ -83,8 +105,8 @@ private class SubscriberInfomation
             {
                 _log.Error(
                     exception,
-                    "Failed to dispatch to event handler {0}",
-                    handler.GetType().PrettyPrint());
+                    $"Failed to dispatch to event handler '{handler.GetType().PrettyPrint()}' for event '{domainEvent.EventType.PrettyPrint()}' due to unexpected exception!");
+                throw;
```

Comment:
@ylbillyli You are right. I was thinking that I wanted to preserve the previous logging behavior, but a cleaner codebase is more favorable